diff_cover/violations_reporter.py

```diff
@@ -107,6 +107,7 @@ def _get_classes(self, xml_document, src_path):
         # within it. If we have that we outta use it
         sources = xml_document.findall('sources')
         sources = [source.find('source').text for source in sources]
+        sources = [source for source in sources if not source == '--source']
```

Comment:
The problem with solving this with a list comprehension is that it has to go over all the sources a second time. So at the very least this should be combined with the line above it

something like
```
source.find('source').text for source in sources if source != '--source']
```

Other than that I wanna see if we can validate the path somehow rather than hard coding ```--source``` I may not be able to do that though. Otherwise we can go with this